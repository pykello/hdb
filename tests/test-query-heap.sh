#!/bin/sh

hdb_port_string=`./launch-hdb.py 2 10 100`
read -a ports <<< $hdb_port_string

echo "Loading ..."
./load-hdb.py ${ports[0]} < ./data/heap.1

echo "Querying ..."
./query-hdb.py ${ports[0]} <<< "=1 * *"
./query-hdb.py ${ports[0]} <<< "=1 * * * *"
./query-hdb.py ${ports[0]} <<< "=1 * * =2 *"
./query-hdb.py ${ports[0]} <<< "=1 * * =3 *"
./query-hdb.py ${ports[0]} <<< "=1 * * * >5"
./query-hdb.py ${ports[0]} <<< "=1 * * * %2"


./stop-hdb.py ${ports[0]}
