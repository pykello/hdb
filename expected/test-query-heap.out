Loading ...
Loaded 6
Querying ...
=1 * *
[2, 3]
=1 * * * *
[4, 5, 6, 7]
=1 * * =2 *
[4, 6]
=1 * * =3 *
[5, 7]
=1 * * * >5
[6, 7]
=1 * * * %2
[4, 6]
Shutting down (localhost, ${ports[0]}) ...
OK.
Shutting down (localhost, ${ports[1]}) ...
OK.
